### PR TITLE
Explain changing GTK theme in FAQ

### PR DESCRIPTION
**Pull request by eht16**
Part of https://github.com/geany/geany/issues/3063.

### CODE REVIEW

**Comment by eht16** [comment]
@techee what to do with the macOS stub? Can the GTK theme changed at all on macOS or is it fixed to "Prof-Gnome"?

**Comment by techee** [comment]
Changing themes is not officially supported on macOS (I create the `settings.ini` file every time Geany is launched and update it based on whether the user has light or dark theme used for macOS with `Prof-Gnome` hard-coded). That said, it is possible to replace the theme inside Geany.app bundle (`Geany.app/Contents/Resources/share/themes/Prof-Gnome`), just the `Prof-Gnome` directory has to be preserved and different theme placed inside it.

I think the appearance is reasonably native-looking though so users shouldn't worry about it much.

**Comment by eht16** [comment]
Thanks.
I added what you said and hope that's still correct :D.
Should we add a note that changes get lost when upgrading Geany on macOS after `Geany.app/Contents/Resources/share/themes/Prof-Gnome` has been changed or will the changes persist?

@elextr would you mind re-reading the new FAQ entry for wording and clarity?

File: page_content/documentation/faq.md [hunk 1]
```diff
@@ -201,6 +201,45 @@ indent settings - i.e. you might want one document indented with tabs whilst
 another is open with spaces indentation. See the [manual][8] for details.
 
 
+## How to change the GTK theme?
+
+The GTK theme defines the look and feel of Geany. Many different themes are available on the net,
+including dark ones. You might have a look at https://www.gnome-look.org/ for GTK3 themes.
```
Comment (elextr) [inline page_content/documentation/faq.md:207]: Suggest "Many different themes, including dark ones, are available on the net, for example from https://www.gnome-look.org/."

No need to mention GTK3, its all Geany does ATM.

File: page_content/documentation/faq.md [hunk 2]
```diff
@@ -201,6 +201,45 @@ indent settings - i.e. you might want one document indented with tabs whilst
...
+
+### On macOS
+
+Geany releases for macOS have the "Prof-Gnome" theme pre-installed and depending on the used macOS theme, the light or dark variant is used automatically.
+
+If desired, the theme can be changed by replacing the folder contents of `Geany.app/Contents/Resources/share/themes/Prof-Gnome` in the Geany.app bundle. Note that the folder name `Prof-Gnome` must be kept and only the contents should be replaced.
+
+### On Linux/Unix systems
+
+Changing the GTK theme on Linux and Unix distributions is done differently depending on the desktop environment you use.
```
Comment (elextr) [inline page_content/documentation/faq.md:240]: Add at the start "Geany will follow the GTK theme set by the normal desktop tools so changing ..."

File: page_content/documentation/faq.md [hunk 3]
```diff
@@ -201,6 +201,45 @@ indent settings - i.e. you might want one document indented with tabs whilst
...
+Note that changing the GTK theme will not affect the syntax highlighting color theme within Geany.
+To change this, see [Color Themes](/download/themes/).
+
+### On Windows
+
+To change the GTK theme, perform the following steps:
+
+- find and download a GTK theme
+- unpack its contents into the directory `c:\Users/<YOUR_USERNAME>\AppData\Local\themes\` (create if it does not exist yet)
+- create the directory `c:\Users/<YOUR_USERNAME>\AppData\Local\gtk-3.0\` if it not already exists
```
Comment (elextr) [inline page_content/documentation/faq.md:218]: " ... if it does not already exist"

File: page_content/documentation/faq.md [hunk 4]
```diff
@@ -201,6 +201,45 @@ indent settings - i.e. you might want one document indented with tabs whilst
...
+  [Settings]
+  gtk-theme-name=Prof-Gnome
+  ```
+- adjust the `gtk-theme-name` setting to match the folder name of the unpacked GTK theme
+
+Themes you may like:
+
+- "Prof-Gnome" https://www.gnome-look.org/p/1334194 (used in Geany for macOS)
+- "Windows 10" https://github.com/B00merang-Project/Windows-10
+- "Windows 10 Dark" https://github.com/B00merang-Project/Windows-10-Dark
```
Comment (elextr) [inline page_content/documentation/faq.md:230]: What about your favourite olde fashioned ... erm I mean "traditional" :smile: one?

File: page_content/documentation/faq.md [hunk 5]
```diff
@@ -201,6 +201,45 @@ indent settings - i.e. you might want one document indented with tabs whilst
...
+
+### On macOS
+
+Geany releases for macOS have the "Prof-Gnome" theme pre-installed and depending on the used macOS theme, the light or dark variant is used automatically.
```
Comment (elextr) [inline page_content/documentation/faq.md:234]: and depending on the macOS theme used, the light or dark variant is used automatically.

File: page_content/documentation/faq.md [hunk 6]
```diff
@@ -201,6 +201,45 @@ indent settings - i.e. you might want one document indented with tabs whilst
 another is open with spaces indentation. See the [manual][8] for details.
 
 
+## How to change the GTK theme?
+
+The GTK theme defines the look and feel of Geany. Many different themes are available on the net,
+including dark ones. You might have a look at https://www.gnome-look.org/ for GTK3 themes.
+
+Note that changing the GTK theme will not affect the syntax highlighting color theme within Geany.
```
Comment (elextr) [inline page_content/documentation/faq.md:209]: Isn't it called a "color scheme" eg `Menu->View->Change Color Scheme...`

**Review by elextr — COMMENTED** [review]
Done

**Comment by elextr** [comment]
Don't forget to link to it from the download page so users will find it.

**Comment by elextr** [comment]
Don't forget to link to it from the download page so users will find it.

**Comment by techee** [comment]
> I added what you said and hope that's still correct :D.

Yep, perfect!

> Should we add a note that changes get lost when upgrading Geany on macOS after  Geany.app/Contents/Resources/share/themes/Prof-Gnome has been changed or will the changes persist?

The changes will get lost, not sure if it's necessary to document it but it depends on the output of @elextr `--pedantic` :-).

**Comment by techee** [comment]
> "Prof-Gnome" https://www.gnome-look.org/p/1334194 (used in Geany for macOS)

By the way the `prof-gnome` version in osx is modified quite a bit from the official release because there were bugs in the theme and I also wanted to make it look more macOS-like. I just pushed a few commits to the geany-osx repo in an attempt to create something that would also look good on Windows.

File: page_content/documentation/faq.md (hunk 4 above)
Comment (techee) [reply page_content/documentation/faq.md:230]: > What about your favourite olde fashioned ... erm I mean "traditional" 😄 one?

I would suggest including that one too. I spent today testing various themes from gnome-look and most didn't work/look very well and the Win95 theme was a pleasant outlier.

Comment (eht16) [reply page_content/documentation/faq.md:230]: What are you talking about? What is "my traditional one"?

File: page_content/documentation/faq.md (hunk 1 above)
Comment (eht16) [reply page_content/documentation/faq.md]: Done

File: page_content/documentation/faq.md (hunk 6 above)
Comment (eht16) [reply page_content/documentation/faq.md]: Done

File: page_content/documentation/faq.md (hunk 3 above)
Comment (eht16) [reply page_content/documentation/faq.md]: Done

File: page_content/documentation/faq.md (hunk 5 above)
Comment (eht16) [reply page_content/documentation/faq.md]: Done

File: page_content/documentation/faq.md (hunk 2 above)
Comment (eht16) [reply page_content/documentation/faq.md]: Done

**Comment by eht16** [comment]
Thanks for the review!

> Don't forget to link to it from the download page so users will find it.

Already done (https://github.com/geany/www.geany.org/pull/40/files#diff-fd05f263f386602f0eb5eedfa0cef4e12ba1625599b97a67c2e9010f9f4ccb4fR12).

> > "Prof-Gnome" https://www.gnome-look.org/p/1334194 (used in Geany for macOS)
> 
> By the way the `prof-gnome` version in osx is modified quite a bit from the official release because there were bugs in the theme and I also wanted to make it look more macOS-like. I just pushed a few commits to the geany-osx repo in an attempt to create something that would also look good on Windows.

Ok, I changed the link to your one.

File: page_content/documentation/faq.md (hunk 4 above)
Comment (techee) [reply page_content/documentation/faq.md:230]: The one from good old times where you could distinguish individual widgets:

https://github.com/grassmunk/Chicago95

This by the way is how the traditional (i.e. non-web-like) user interface widgets look on Windows 11:
1. Task one - notice there are tabs
2. Task two - tell which one is selected

Crazy UI design.

image

Comment (eht16) [reply page_content/documentation/faq.md:230]: Ok, I just added it. I still can't really take this theme seriously hence my confusion :D.

**Pull request merged**